client/apps/edit/components/content/sections/related_articles/index.jsx

```diff
@@ -0,0 +1,151 @@
+import async from 'async'
+import request from 'superagent'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { compact, pluck, without } from 'underscore'
+import { data as sd } from 'sharify'
+import { EditArticleCard } from './components/edit_article_card'
+import { RelatedArticlesInput } from './components/related_articles_input'
+import { ArticleCard } from '@artsy/reaction-force/dist/Components/Publishing/Series/ArticleCard'
+import DraggableList from '../../../../../../components/drag_drop/index.coffee'
+
+export class RelatedArticles extends Component {
+  static propTypes = {
+    article: PropTypes.object.isRequired,
+    color: PropTypes.string,
+    onChange: PropTypes.func
+  }
+
+  state = {
+    relatedArticles: [],
+    loading: true
+  }
+
+  componentWillMount = () => {
+    this.fetchArticles()
+  }
+
+  fetchArticles = () => {
+    const { related_article_ids } = this.props.article.attributes
+    let relatedArticles = []
+
+    if (related_article_ids && related_article_ids.length) {
+      return async.each(related_article_ids, (id, cb) => {
+        return request
+          .get(`${sd.API_URL}/articles/${id}`)
+          .set({'X-Access-Token': (sd.USER != null ? sd.USER.access_token : undefined)})
+          .end((err, res) => {
+            if (err) {
+              console.log(err)
+            }
+            relatedArticles.push(res.body)
+            cb()
+          })
+      }
+      , () => {
+        this.setState({
+          loading: false,
+          relatedArticles: compact(relatedArticles)
+        })
+      })
+    } else {
+      this.setState({
+        loading: false,
+        relatedArticles
+      })
+    }
+  }
+
+  onAddArticle = (related_article_ids) => {
+    const { onChange } = this.props
+
+    onChange('related_article_ids', related_article_ids)
+    this.fetchArticles()
```

Comment:
Interesting, is it necessary to fetch all the articles each time one is added or removed? Ideally we should be able to get the data we need to render the card from the `RelatedArticlesInput` autocomplete so it seems like this additional fetch is redundant.

I see you've added a new version of `Autocomplete` that is designed for an array of `ids` that this component uses though. I think we could rework it a bit to leverage more from the fetch. I don't know how easy this change would be and I'm definitely not asking to make this change now, but this does feel like a worthy optimization for the future!

Reply:
Yeah not ideal to fetch all all the time -- and definitely don't need to fetch on removing.

Reply:
This fetch is now updated to use graphql and not repeat itself